### DISCUSSION TITLE
The java language server is a bit problematic on windows.

### DISCUSSION THREAD

**Discussion by pathwave** [helix-editor/helix#5861] [General]
The file to start Java Eclipse JDT Language Server "jdtls" is a python script and not a binary / executable file.
Which means for helix to start it it would actually need to start python with the path to jdtls as an argument 
(i.e language-server = { command = "python", args = ["path/to/jdtls"]} ), which is problematic since python wont look in $PATH for the argument. (and also it requires python to be installed)

Alternatively it can be started with java but this is also problematic as the java file has version number in its name.

(The vscode extension avoids this problem since it installs a specific version of the language server and java)

 ¯\\\_(ツ)_/¯

**Comment by Proful** [comment]
I am facing the same issue on Windows with jlts.

Please let me know what are the steps you are trying, so that I can try that as well. And update here.

Jlts configuration feels bit intimidating.

**Comment by cstich** [comment]
I do not have a Windows machine to try this on, but what's the exact problem to starting the LSP with java? Does something like this not work on Windows?

```
java \
        -Declipse.application=org.eclipse.jdt.ls.core.id1 \
	-Dosgi.bundles.defaultStartLevel=4 \
	-Declipse.product=org.eclipse.jdt.ls.core.product \
	-Dlog.level=ALL \
	-noverify \
	-Xmx1G \
	--add-modules=ALL-SYSTEM \
	--add-opens java.base/java.util=ALL-UNNAMED \
	--add-opens java.base/java.lang=ALL-UNNAMED \
	-jar ./plugins/org.eclipse.equinox.launcher_1.5.200.v20180922-1751.jar \
	-configuration ./config_linux \
	-data /path/to/data`

**Reply by pathwave** [reply]
It works, the problem here is that this needed a custom language file with a hard-coded path to the .jar file. So the point of the issue is about finding a way that helix could automatically start the LSP, so that it works "out-of-the-box".

In other words this argument: "-jar ./plugins/org.eclipse.equinox.launcher_1.5.200.v20180922-1751.jar" the path depends on where the LSP is installed and what version of the LSP is installed, and we cant use the PATH variable to get that information :/

**Comment by dead10ck** [comment]
Why can't you just run `jdtls` without the explicit `python` invocation?

**Comment by lesleyrs** [comment]
Created PR here https://github.com/eclipse-jdtls/eclipse.jdt.ls/pull/2808

**Reply by lesleyrs** [reply]
It's merged @pathwave

It still requires python, but that's necessary to keep the same command as on Linux and to have it work out of the box.
